### PR TITLE
Qingying model router streamlit

### PR DESCRIPTION
**Pull request by qmeng222**
Add GGUF model router support across all model types (NLP, Vision, Multimodal, and Audio) in Streamlit.

### CODE REVIEW

**Comment by unseenmars** [comment]
@zhiyuan8 This is the step one of the model router for Streamlit interface.

File: nexa/gguf/streamlit/streamlit_image_chat.py [hunk 1]
```diff
@@ -1,29 +1,129 @@
...
+from nexa.general import pull_model
 from nexa.gguf.nexa_inference_image import NexaImageInference
 import io
 
-default_model = sys.argv[1]
-is_local_path = False if sys.argv[2] == "False" else True
-hf = False if sys.argv[3] == "False" else True
+# init:
+DEFAULT_PARAMS = {
+    "num_inference_steps": 50,
```
Comment (zhiyuan8) [inline nexa/gguf/streamlit/streamlit_image_chat.py]: 50 steps is too slow, and lcm-dreamshaper only takes 4 steps, sd1.5 takes 20 steps
https://github.com/NexaAI/nexa-sdk/blob/main/nexa/gguf/nexa_inference_image.py#L176

also, image generation parameters are predefined here
https://github.com/NexaAI/nexa-sdk/blob/main/nexa/constants.py#L286-L320

**Review by zhiyuan8 — CHANGES_REQUESTED** [review]

File: nexa/gguf/streamlit/streamlit_image_chat.py [hunk 2]
```diff
@@ -1,29 +1,129 @@
...
+    "height": 512,
+    "width": 512,
+    "guidance_scale": 7.5,
+    "random_seed": 42
+}
 
+def parse_nexa_list_output() -> List[str]:
```
Comment (zhiyuan8) [inline nexa/gguf/streamlit/streamlit_image_chat.py:26]: why not directly get add a helper funciton here? running "nexa list" with string parsing is not a good practice

https://github.com/NexaAI/nexa-sdk/blob/6b30851fdbe6e263531778e15638a283cd4ed5c1/nexa/general.py#L510-L517

File: nexa/gguf/streamlit/streamlit_vlm.py [hunk 3]
```diff
@@ -1,40 +1,108 @@
...
+try:
+    default_model = sys.argv[1]
+    is_local_path = sys.argv[2].lower() == "true"
+    hf = sys.argv[3].lower() == "true"
+    projector_local_path = sys.argv[4] if len(sys.argv) > 4 else None
+except IndexError:
+    st.error("Missing required command line arguments.")
+    sys.exit(1)  # program terminated with an error
 
+def parse_nexa_list_output() -> List[str]:
```
Comment (zhiyuan8) [inline nexa/gguf/streamlit/streamlit_vlm.py:22]: same issue, do not parse string
also for general functions, put them in utils.py and import, do not copy code blocks multiple times.

File: nexa/gguf/streamlit/streamlit_vlm.py [hunk 4]
```diff
@@ -1,40 +1,108 @@
...
+                            break
+                else:
+                    vlm_models.add(model_name)
 
+        return sorted(list(vlm_models))  # convert to sorted list for consistent ordering
+    except Exception as e:
+        st.error(f"Error parsing nexa list output: {str(e)}")
+        return []
+
+def get_model_options() -> List[str]:
```
Comment (zhiyuan8) [inline nexa/gguf/streamlit/streamlit_vlm.py:52]: same, do not hard copy the same function multiple places

File: nexa/gguf/streamlit/streamlit_vlm.py [hunk 5]
```diff
@@ -1,40 +1,108 @@
...
+    """Get list of model options from nexa list."""
+    models = parse_nexa_list_output()
+    # add special options at the end of the dropdown menu:
+    models.extend(["Use Model From Nexa Model Hub 🔍", "Local Model 📁"])
+    return models
+
+def update_model_options():
```
Comment (zhiyuan8) [inline nexa/gguf/streamlit/streamlit_vlm.py:59]: same, do not hard copy the same function multiple places

File: nexa/gguf/streamlit/streamlit_image_chat.py [hunk 6]
```diff
@@ -1,29 +1,129 @@
...
+        return []
+
+def get_model_options() -> List[str]:
+    """Get list of model options from nexa list."""
+    models = parse_nexa_list_output()
+    # add special options at the end of the dropdown menu:
+    models.extend(["Use Model From Nexa Model Hub 🔍", "Local Model 📁"])
+    return models
+
+def update_model_options():
```
Comment (zhiyuan8) [inline nexa/gguf/streamlit/streamlit_image_chat.py:53]: same issue, no hard copy

**Review by zhiyuan8 — CHANGES_REQUESTED** [review]
same functions are copied multiple times, need code refactoring.

**Pull request closed without merging**